### PR TITLE
Add erase_args parameter to add_logging_args()

### PR DESCRIPTION
**Pull request by irinakhismatullina**
Implemented [this suggestion](https://github.com/src-d/style-analyzer/pull/691#discussion_r264763517).

### CODE REVIEW

File: modelforge/slogging.py [hunk 1]
```diff
@@ -219,11 +219,13 @@ def set_context(context):
...
 
 
-def add_logging_args(parser: argparse.ArgumentParser, patch: bool = True) -> None:
+def add_logging_args(parser: argparse.ArgumentParser, patch: bool = True,
+                     erase_args: bool = True) -> None:
     """
     Add command line flags specific to logging.
 
     :param parser: `argparse` parser where to add new flags.
+    :param erase_args: Automatically erase logging arguments after setup.
```
Comment (vmarkovtsev) [inline modelforge/slogging.py]:
```suggestion
    :param erase_args: Automatically erase logging-related flags after parsing.
```

File: modelforge/slogging.py [hunk 2]
```diff
@@ -219,11 +219,13 @@ def set_context(context):
...
 
 
-def add_logging_args(parser: argparse.ArgumentParser, patch: bool = True) -> None:
+def add_logging_args(parser: argparse.ArgumentParser, patch: bool = True,
+                     erase_args: bool = True) -> None:
     """
     Add command line flags specific to logging.
 
     :param parser: `argparse` parser where to add new flags.
+    :param erase_args: Automatically erase logging-related flags after parsing.
```
Comment (vmarkovtsev) [inline modelforge/slogging.py]:
```suggestion
    :param erase_args: Automatically remove logging-related flags from parsed args.
```

**Comment by vmarkovtsev** [comment]
@irinakhismatullina Tests failure

**Comment by irinakhismatullina** [comment]
On it

**Comment by irinakhismatullina** [comment]
@vmarkovtsev Ready. Please check the test code, i'm not shure that 'pass' means everything is 100% correct.

File: modelforge/tests/test_logs.py [hunk 3]
```diff
@@ -48,6 +48,14 @@ def test_argparse_no_patch(self):
         slogging.add_logging_args(parser)
         self.assertNotEqual(method, parser.parse_args)
 
+    def test_argparse_erase_args(self):
+        for var in ("2", "3"):
```
Comment (vmarkovtsev) [inline modelforge/tests/test_logs.py]: No need to go so complex. The reason why I had to run a separate process is that logging setup cannot be reverted forever.

In your case it is enough to patch `sys.args`, patch `slogging.setup()` to be a no-op, check the parsed args, restore `slogging.setup()`, restore `sys.args`

Comment (vmarkovtsev) [reply modelforge/tests/test_logs.py]: `unittest.patch` is your friend

**Comment by vmarkovtsev** [comment]
Now is awesome 👍

**Comment by vmarkovtsev** [comment]
But test-failing 😭

**Comment by irinakhismatullina** [comment]
@vmarkovtsev Finally

**Pull request merged**